common/key_value_database/platform/lmdb/lmdb.cpp

```diff
@@ -118,7 +119,7 @@ expected::ExpectedBytes KeyValueDatabaseLmdb::Read(const string &key) {
 		}
 	});
 	if (err) {
-		return err;
+		return tl::make_unexpected(err);
```

Comment:
IMHO this defeats some of the convenience we already have. I also don't fully understand why it's needed. Isn't `tl::expected<Type1, Type2>(Type2 &)` a valid constructor?

Reply:
Hmm, yes, looks like it. I understand it from their perspective. Our types are not easily confused, but that may not be the case for everyone. A bit annoying, but I think it is an acceptable price to pay.

Reply:
Yeh, we can create a conversion operator for this though...

_if we want to_...